Engine/Helper.cs

```diff
@@ -1323,156 +1323,168 @@ internal List<RuleSuppression> GetSuppressionsConfiguration(ConfigurationDefinit
         /// Suppress the rules from the diagnostic records list.
         /// Returns a list of suppressed records as well as the ones that are not suppressed
         /// </summary>
-        /// <param name="ruleSuppressions"></param>
-        /// <param name="diagnostics"></param>
         public Tuple<List<SuppressedRecord>, List<DiagnosticRecord>> SuppressRule(
             string ruleName,
             Dictionary<string, List<RuleSuppression>> ruleSuppressionsDict,
             List<DiagnosticRecord> diagnostics,
-            out List<ErrorRecord> errorRecords)
+            out List<ErrorRecord> errors)
         {
-            List<SuppressedRecord> suppressedRecords = new List<SuppressedRecord>();
-            List<DiagnosticRecord> unSuppressedRecords = new List<DiagnosticRecord>();
-            Tuple<List<SuppressedRecord>, List<DiagnosticRecord>> result = Tuple.Create(suppressedRecords, unSuppressedRecords);
-            errorRecords = new List<ErrorRecord>();
-            if (diagnostics == null || diagnostics.Count == 0)
+            var suppressedRecords = new List<SuppressedRecord>();
+            var unsuppressedRecords = new List<DiagnosticRecord>();
+            errors = new List<ErrorRecord>();
+            var result = new Tuple<List<SuppressedRecord>, List<DiagnosticRecord>>(suppressedRecords, unsuppressedRecords);
+
+            if (diagnostics is null || diagnostics.Count == 0)
             {
                 return result;
             }
 
-            if (ruleSuppressionsDict == null || !ruleSuppressionsDict.ContainsKey(ruleName)
-                || ruleSuppressionsDict[ruleName].Count == 0)
+            if (ruleSuppressionsDict is null
+                || !ruleSuppressionsDict.TryGetValue(ruleName, out List<RuleSuppression> ruleSuppressions)
+                || ruleSuppressions.Count == 0)
             {
-                unSuppressedRecords.AddRange(diagnostics);
+                unsuppressedRecords.AddRange(diagnostics);
                 return result;
             }
 
-            List<RuleSuppression> ruleSuppressions = ruleSuppressionsDict[ruleName];
-            var offsetArr = GetOffsetArray(diagnostics);
-            int recordIndex = 0;
-            int startRecord = 0;
-            bool[] suppressed = new bool[diagnostics.Count];
-            foreach (RuleSuppression ruleSuppression in ruleSuppressions)
+            // We need to report errors for any rule suppressions with IDs that are not applied to anything
```

Comment:
I think I get why we need to report on suppression's that were not applied, but I'm not sure why that's an error. Could you elaborate in the comment?